include/rcutils/error_handling.h

```diff
@@ -89,6 +89,35 @@ void
 rcutils_set_error_state(
   const char * error_msg, const char * file, size_t line_number, rcutils_allocator_t allocator);
 
+/// Check an argument for a null value.
+/**
+ * If the argument's value is null, set the error message saying so and
+ * return the error_return_type after setting the error message.
+ *
+ * \param[in] argument The argument to test.
+ * \param[in] error_return_type The type to return in case of an error.
+ * \param[in] allocator The allocator to use if an error message needs to be allocated.
+ */
+#define RCUTILS_CHECK_ARGUMENT_FOR_NULL(argument, error_return_type, allocator) \
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(argument, #argument " argument is null", \
+    return error_return_type, allocator)
+
+/// Check an argument for a null value, with a user defined message and error statement.
+/**
+ * If the argument's value is null, set the error statement will be evaluated
+ * after setting the error message.
+ *
+ * \param[in] value The value to test.
+ * \param[in] msg The error message if value is false.
+ * \param[in] error_statement The statement to evaluate if value is false.
+ * \param[in] allocator The allocator to use if an error message needs to be allocated.
+ */
+#define RCUTILS_CHECK_FOR_NULL_WITH_MSG(value, msg, error_statement, allocator) \
+  if (!(value)) { \
+    RCUTILS_SET_ERROR_MSG(msg, allocator); \
```

Comment:
should the allocator be checked before calling SET_ERROR_MSG? An alternative would be to check it in `rcutils_set_error_state` itself

Reply:
There's no need to check as it's a pass by value at the moment.

At a high level do we want to be passing `allocator_t` or `allocator_t *`s The current implementation is not using the pointers. And I missed that in [your pr](https://github.com/ros2/rcl/pull/178) which assumes pointers. 

The current implementations almost always do not use the pointer. I'd originally implemented things using the pointer but switched to pass by value to homogenize with the current implementation.

Reply:
Sorry rereading this I was distracted by the other commit. We should definitely check in one of the two places. Since rcutils_set_error_state is public I think we have to check at that level. And since this function won't error with it not set it might be ok to pass through w/o checking.

Reply:
Sounds good to me I'll add a check in the rcutils function `rcutils_set_error_state`. This should not block this PR as this one just keeps the current behavior and the rcutils one will be an improvement.